online-shopping-website/src/Components/ProductGrid.js

```diff
@@ -3,47 +3,55 @@ import ButtonBase from '@mui/material/ButtonBase';
 import Grid from '@mui/material/Grid';
 import React from 'react';
 import Stack from '@mui/material/Stack';
+import {ProductDetails} from './ProductDetails';
+import {BrowserRouter, Outlet, Link, Routes, Route} from "react-router-dom";
 
 const ProductPreview = (props) => {
-  return (
-    <ButtonBase sx={{ width: '100%', textAlign: 'left', margin: 0, padding: 0, borderRadius: '15px'}} >
-      <Card className="ProductContainer">
-        <Stack sx={{ height: '100%' }} direction="column" justifyContent="space-between">
-          {/* Image */}
-          <Stack direction="row" justifyContent="center" sx={{ textAlign: 'center' }}>
-            <img className="ProductImage" src={props.product.image} alt={props.product.name} />
-          </Stack>
+    return (
+        // Navigates to ProductDetails page on click
+        <Link to={{
+            pathname: `/${props.product.name}`,
+            params: {props}
+        }}>
```

Comment:
Surrounding the Button in `<Link>` seems to change the styling of the text in the buttons. It creates blue underlines under all text in the button. Try to find a way to insert an onClick or something to the button, and handle the link there.